src/components/generics/vacation-test-forms/vacationRequestForm.tsx

```diff
@@ -0,0 +1,217 @@
+import { CalendarPickerView } from "@mui/x-date-pickers";
+import { Box, Button, FormControl, InputLabel, MenuItem, Select, SelectChangeEvent, TextField, Typography } from "@mui/material";
+import { VacationType } from "generated/client";
+import strings from "localization/strings";
+import { ChangeEvent, useEffect, useState } from "react";
+import { FilterScopes, VacationData } from "types";
+import useEditorContentStyles from "styles/editor-content/editor-content";
+import Holidays from "date-holidays";
+import DateRangePicker from "components/generics/date-range-picker/date-range-picker";
+
+interface VacationRequestFormProps {
+  onClick: () => void;
+  buttonLabel: string;
+  vacationData: VacationData;
+  setVacationData: (vacationData: VacationData) => void;
+}
+
+/**
+* Form component for vacation requests
+* @param param0 
+*/
+const VacationRequestForm = ({
+  onClick,
+  buttonLabel,
+  vacationData,
+  setVacationData
+}: VacationRequestFormProps) => {
+  const classes = useEditorContentStyles();
+  const dateFormat = "yyyy.MM.dd";
+  const [ datePickerView ] = useState<CalendarPickerView>("day");
+
+  /**
+ * handle DateRangePicker chances
+ * 
+ * @param date
+ * @param isStart
+ */
+  const handleVacationDateChange = (date: Date | null, isStart: boolean) => {
+    if (!date) return;
+    if (isStart) {
+      const newRequest: VacationData = { ...vacationData, startDate: date };
+      setVacationData(newRequest);
```

Comment:
Could this be done in a single line e.g.
setVacationData({ ...vacationData, startDate: date });